### PR TITLE
percentiles calculated and pushed

### PR DESCRIPTION
**Pull request by kjgross**

### CODE REVIEW

File: public/javascripts/quiz.js [hunk 1]
```diff
@@ -88,9 +88,8 @@ var uploadScore = function(correct_answers){
         success: function(response){
           console.log(response);
           var allScores = response.all_scores,
-          thisQuizScore = response.this_score;
-          console.log("allScores"+allScores);
-          console.log("thisquiz"+thisQuizScore);
+          thisQuizScore = response.this_score,
```
Comment (blakeparkinson) [inline public/javascripts/quiz.js]: 4 sure gonna be a merge conflict :(

File: routes/quiz.js [hunk 2]
```diff
@@ -73,6 +73,9 @@ router.get('/results', function(req, res) {
     else {
       modified_score = quiz_score / 3
     }
+    mod_scores.push(modified_score); // add this quiz's score to the all scores array
```
Comment (blakeparkinson) [inline routes/quiz.js]: where is mod_scores ever getting more than the single value that is coming right here?

Comment (kjgross) [inline routes/quiz.js]: in the fetchQuizScores. in the background on the quiz page, I pull all the old scores from mongo. Then, once I get the result from this quiz, I append it in, sort it, and call the percentile fxn on it

File: routes/quiz.js [hunk 3]
```diff
@@ -83,17 +86,39 @@ router.get('/results', function(req, res) {
...
         }
         else {
-          res.json({all_scores: req.session.scores, this_score: modified_score, success: true});
+          res.json({all_scores: req.session.scores, this_score: modified_score, success: true, percentile: req.session.percentile});
         }
       });
     });
 });
 
+/* Takes in a sorted array that includes the score and the score. Sorted position over length = percentile.
```
Comment (blakeparkinson) [inline routes/quiz.js]: the first comment in this sentence doesnt make sense

File: routes/quiz.js [hunk 4]
```diff
@@ -83,17 +86,39 @@ router.get('/results', function(req, res) {
...
     });
 });
 
+/* Takes in a sorted array that includes the score and the score. Sorted position over length = percentile. 
+Am ignoring true math piece of even/odd stuff */
+var calculatePercentile = function(req, score, all_scores){
+  len = all_scores.length;
+  count = 0;
+  for (i=0;i<all_scores.length;i++){
+    if(all_scores[i]< score){
```
Comment (blakeparkinson) [inline routes/quiz.js]: what is the structure of all_scores? maybe a comment would help, but i dont really understand why we ++ if it's less than, return if its == or > than.

File: routes/quiz.js [hunk 5]
```diff
@@ -83,17 +86,39 @@ router.get('/results', function(req, res) {
...
+      count++
+    }
+    else if (all_scores[i] == score){
+      percentile = (count/len)*100
+      req.session.percentile = percentile
+      return;
+    }
+    else if (all_scores[i] > score){
```
Comment (blakeparkinson) [inline routes/quiz.js]: this can just be an else with a comment that it will be greater than

**Comment by blakeparkinson** [comment]
yay :tiger:

**Pull request merged**